higher_health/views.py

```diff
@@ -72,29 +85,79 @@ def get_initial(self):
         return initial_data
 
 
-def healthcheck_receipt(request):
-    try:
-        if "triage_id" in request.session:
-            triage = Covid19Triage.objects.get(id=request.session["triage_id"])
-            data = {
-                "risk_level": triage.risk,
-                "first_name": triage.first_name,
-                "last_name": triage.last_name,
-                "timestamp": triage.timestamp,
-                "msisdn": triage.hashed_msisdn,
-            }
-            return render(request, "healthcheck_receipt.html", data)
-    except Covid19Triage.DoesNotExist:
-        pass
+def staff_check(user):
+    return not user.is_staff
+
 
+@login_required
+@user_passes_test(staff_check)
+def healthcheck_receipt(request):
+    triage = Covid19Triage.objects.filter(user=request.user).last()
+    if triage:
+        data = {
+            "risk_level": triage.risk,
+            "first_name": triage.first_name,
+            "last_name": triage.last_name,
+            "timestamp": triage.timestamp,
+            "msisdn": triage.hashed_msisdn,
+        }
+        return render(request, "healthcheck_receipt.html", data)
     return HttpResponseRedirect("/")
 
 
-def healthcheck_login(request):
-    form = HealthCheckLogin()
-    return render(request, "healthcheck_receipt.html", {"form": form})
+class HealthCheckLoginView(generic.FormView):
+    form_class = HealthCheckLogin
+    template_name = "login.html"
+    success_url = reverse_lazy("healthcheck_otp")
+
+    def send_otp_sms(self, user):
+        otp = "".join(secrets.choice(string.digits) for _ in range(6))
+        h = hmac.new(otp.encode(), digestmod=sha256)
```

Comment:
I'm wondering if, instead of using the OTP as the key, and having a blank body, we use the django secret key as the key, and the OTP as the body. 

That way we're the only ones that can generate valid hashes, since we're the only ones who have the secret key.

I think otherwise it will be trivial for me to generate all 100,000 possible hashes on my local machine, then I can easily reverse the hash.

With a secret, I would need to know the secret before I can reverse the hash.

Reply:
@rudigiesler Agreed. I've made the change